scipy/optimize/_differentialevolution.py

```diff
@@ -193,6 +194,28 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
 
         .. versionadded:: 1.7.0
 
+    integrality : 1-D array, optional
+        For each decision variable, a boolean value indicating whether the
+        decision variable is constrained to integer values. The array is
+        broadcast to ``(len(x),)``.
+        If any decision variables are constrained to be integral, polishing
+        will switched off (e.g. ``polish=False``).
+        The solution vectors are passed to the objective function as
+        ``func(np.round(x[integrality]))``.
+        For variables considered as integers consider the bounds carefully.
```

Comment:
It seems that this could be done automatically, right?

Reply:
Yes, it can be done automatically. BUT.....
The user would have to be informed that the minimiser has altered the limits to what was originally passed.
e.g. `[(np.round(lb) - (0.5 - ULP), np.round(ub) + (0.5 - ULP)]`.

It seems unnatural to me to change the bounds that the user provided.

Reply:
You're not really changing their bounds, though. You still respect the user's bounds exactly. You're only doing this to give the algorithm the change of selecting those integer values that it deserves, right?